.github/workflows/cpp-workflow.yml

```diff
@@ -0,0 +1,332 @@
+      matrix:
+        sub_component: ${{ fromJSON(inputs.components) }}
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.cloud.mov.ai/devops/ros-buildtools-noetic:latest
+      options: --user root
+      credentials:
+        username: ${{secrets.registry_user}} 
+        password: ${{secrets.registry_password}}
+
+    steps:
+    - name: Checkout    
+      uses: actions/checkout@v2
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Find Package details
+      id: vars
+      run: |
+        PACKAGE_VERSION=$(dpkg-parsechangelog --show-field Version)
+        PACKAGE_NAME=$(dpkg-parsechangelog --show-field Source)
+        echo ::set-output name=pkg_name::$PACKAGE_NAME
+        echo ::set-output name=pkg_version::$PACKAGE_VERSION
+      working-directory: ${{ matrix.sub_component }}
+
+    - name: Raise Version
+      id: vars-after-raise
+      run: |
+        PACKAGE_VERSION=${{ steps.vars.outputs.pkg_version }}
+        main_version=${PACKAGE_VERSION%-*}
+        build_id="${PACKAGE_VERSION#*-}"
+        raised_buildid=$(($build_id+1))
+        RAISED_PACKAGE_VERSION="$main_version-$raised_buildid"
+        echo ::set-output name=pkg_version::$RAISED_PACKAGE_VERSION
+      working-directory: ${{ matrix.sub_component }}
+
+    - name: Install requirements
+      shell: bash
+      run: |
+        IFS=';' read -r -a array <<< '${{ inputs.custom_requirements }}'
+
+        for cmd in "${array[@]}"
+        do
+            echo "$cmd"
+            eval $cmd
+        done
+        echo "finished custom script, now the requirement.apt"
+        apt-get install $(cat ./requirements.apt) -y
+      working-directory: ${{ matrix.sub_component}}
+ 
+        #          curl -fsSL http://packages.osrfoundation.org/gazebo.key | apt-key add - ;add-apt-repository "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main"
+
+
+    - name: SonarQube Scan
+      uses: SonarSource/sonarqube-scan-action@v1.0.0
+      env:
+        SONAR_TOKEN: ${{ secrets.sonar_token }}
+        SONAR_HOST_URL: https://sonarqube.aws.cloud.mov.ai
+      with:
+        projectBaseDir: ./${{ matrix.sub_component }}
+        args: >
+          -Dsonar.organization=MOV-AI
+          -Dsonar.projectKey=${{ github.event.repository.name }}
+          -Dsonar.verbose=true
+          -Dsonar.sources=.
+          -Dsonar.scm.provider=git
+          -Dsonar.qualitygate.wait=true
+          -Dsonar.qualitygate.timeout=300
+
+    - name: Link to SonarQube Dashboard
+      shell: bash
+      run: |
+        echo "Please check report here: https://sonarqube.aws.cloud.mov.ai/dashboard?id=${{ github.event.repository.name }}"
+      id: sonar_links
+
+    
+    - name: Build
+      run: |
+        mkdir build
+        cd build
+        cmake .. -DCMAKE_PREFIX_PATH=/usr/lib/x86_64-linux-gnu/
```

Comment:
here we should have some arguments / env vars to customise the build, for example in https://github.com/marketplace/actions/build-cmake they present:

'''
          build-dir: ${{ runner.workspace }}/build
          # will set the CC & CXX for cmake
          cc: gcc
          cxx: g++
          build-type: Release
          # Extra options pass to cmake while configuring project
          configure-options: -DCMAKE_C_FLAGS=-w32 -DPNG_INCLUDE=OFF
          run-test: true
          ctest-options: -R mytest
          # install the build using cmake --install
          install-build: true
          # run build using '-j [parallel]' to use multiple threads to build
          parallel: 14
'''
or here https://github.com/marketplace/actions/build-c-project